### ISSUE TITLE
Methods like sendResponse() should contain never-return typehint

### ISSUE THREAD

**Issue by janbarasek** [nette/application#290]
Hello,

I think methods like `sendJson()` should contain PHPStan annotation `@phpstan-return never-return` like this:

```php
/**
 * Sends JSON data to the output.
 * @throws Nette\Application\AbortException
 * @phpstan-return never-return
 */
public function sendJson(mixed $data): void
{
	$this->sendResponse(new Responses\JsonResponse($data));
}
```

Currently, this can generate hundreds of errors when PhpStan reports possible undefined variables. The annotation marks the code after this call as dead.

For example:

```php
public function actionDefault(): void
{
    if ($this->variable) {
        $a = 1;
    } else {
        $this->sendJson('error');
    }

    echo $a; // generate possible undefined variable, but it is not true
}
```

Yes, there is an official extension for PhpStan and Nette framework that does this solution, but it would be good to have native support as well.

Thanks!

**Comment by JanTvrdik** [comment]
The official type name is just `never`.
https://github.com/php/php-src/pull/6761

**Comment by dg** [comment]
`@return never` seems good, can you send PR?

**Comment by janbarasek** [comment]
Solved by #291
